tests/unit/test_services.py

```diff
@@ -32,6 +33,23 @@
 ]
 
 
+def test_fix_files_ignore_shebang() -> None:
+    """Ignores shebang lines if present at the beginning of the source."""
+    source = dedent(
+        """\
+        #! /this/line/should/be/ignored
+        ---
+        program: yamlfix
+        """
+    )
+
+    fakestdin = StringIO(source)
+    fakestdin.name = "<stdin>"
+    result = fix_files((fakestdin,))
```

Comment:
`fix_files` is a wrapper function to run `fix_code` for all the files it receives as argument. Therefore, if you test `fix_code` you're testing that part of `fix_files`.

Given that the setup of the tests of `fix_code` is more clear, I'd use it instead. 

So my suggestion would be to have in this test:

```python
    source = dedent(
        """\
        #! /this/line/should/be/ignored
        ---
        program: yamlfix
        """
    )

    result = fix_code(source)

    assert result == source
    ```

Reply:
I respectfully disagree, because my fix for the shebang line problem wasn't implemented in fix_code, but in fix_files. So calling fix_code from the test code to check my patch, which is to fix_files, will always return a failure.

I think it was more straightforward to patch fix_files than fix_code, and I believe it's also the case wrt issue #135

Reply:
I see what you mean. There's another reason why I think it's better to implement this functionality at `fix_code` level. The program is meant to be used as a library too, and a user may have some string that want's to fix, which doesn't belong from a file. To be able to benefit from your feature, they would need to save that string in a file, and then run `fix_files` on it.

Please take a moment to think about it, and if you still think it's not worth it, I'll resolve the conversation and merge the PR.

Sorry the PR is taking so long to be resolved, I'm having some busy days and weren't able to review this sooner :(

Reply:
OK I'll move the fix to fix_code instead of fix_files and modify the test accordingly. I'll try too do this tomorrow.

Reply:
Thank you for the comprehension and the extra work